packages/decorators/src/djs-decorators.ts

```diff
@@ -42,38 +59,111 @@ const DMAvailablePermissions = new Permissions(
+			});
 		}
 
 		if (isGuildBasedChannel(message.channel)) {
 			const missingPermissions = message.channel.permissionsFor(message.guild!.me!)!.missing(resolved);
 
 			if (missingPermissions.length) {
 				throw new UserError({
-					identifier: DecoratorIdentifiers.RequiresPermissionsMissingPermissions,
+					identifier: DecoratorIdentifiers.RequiresClientPermissionsMissingPermissions,
+					message: `Sorry, but I am not allowed to do that. I am missing the permissions: ${missingPermissions}`,
+					context: {
+						missing: missingPermissions
+					}
+				});
+			}
+		}
+
+		return true;
+	});
+};
+
+/**
+ * Allows you to set permissions required for individual methods. This is particularly useful for subcommands that require specific permissions.
+ * @remark This decorator applies to the user of the command. For setting permissions required for the client see {@link RequiresClientPermissions}
+ * @remark This decorator makes the decorated function asynchronous, so any result should be `await`ed.
+ * @param permissionsResolvable Permissions that the method should have.
+ * @example
+ * ```typescript
+ * import { ApplyOptions, RequiresUserPermissions } from '@sapphire/decorators';
+ * import { SubCommandPluginCommand, SubCommandPluginCommandOptions } from '@sapphire/plugin-subcommands';
+ * import type { Message } from 'discord.js';
+ *
+ * (at)ApplyOptions<SubCommandPluginCommandOptions>({
+ * 	aliases: ['cws'],
+ * 	description: 'A basic command with some subcommands',
+ * 	subCommands: ['add', 'remove', 'reset', { input: 'show', default: true }]
+ * })
+ * export default class extends SubCommandPluginCommand {
+ *     // Anyone should be able to view the result, but not modify
+ * 	public async show(message: Message) {
+ * 		return message.channel.send('Showing!');
+ * 	}
+ *
+ * 	(at)RequiresUserPermissions('BAN_MEMBERS') // This subcommand requires the user of the command to be able to ban members.
+ * 	public async add(message: Message) {
+ * 		return message.channel.send('Adding!');
+ * 	}
+ *
+ * 	(at)RequiresUserPermissions('BAN_MEMBERS') // This subcommand requires the user of the command to be able to ban members.
+ * 	public async remove(message: Message) {
+ * 		return message.channel.send('Removing!');
+ * 	}
+ *
+ * 	(at)RequiresUserPermissions('BAN_MEMBERS') // This subcommand requires the user of the command to be able to ban members.
+ * 	public async reset(message: Message) {
+ * 		return message.channel.send('Resetting!');
+ * 	}
+ * }
+ * ```
+ */
+export const RequiresUserPermissions = (...permissionsResolvable: PermissionResolvable[]): MethodDecorator => {
+	const resolved = new Permissions(permissionsResolvable);
+	const resolvedIncludesServerPermissions = Boolean(resolved.bitfield & DMAvailableUserPermissions.bitfield);
+
+	return createFunctionPrecondition((message: Message) => {
+		if (isDMChannel(message.channel) && resolvedIncludesServerPermissions) {
```

Comment:
Faster checks first:
```suggestion
		if (resolvedIncludesServerPermissions && isDMChannel(message.channel)) {
```